mitodl/ocw-hugo-themes
# course card line clamp

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/746

#### What's this PR do?
This PR introduces a new `line-clamp` mixin that can be used to clamp an arbitrary amount of lines at any given line height with any custom amount of extra padding.  This new mixin was applied to the title, instructors and topics in course carousel cards to prevent text from overflowing the static height of the cards.  

#### How should this be manually tested?
 - Clone https://github.mit.edu/ocw-content-rc/ocw-www
 - Set the following values in your `.env`:
```
OCW_STUDIO_BASE_URL=https://ocw-studio-rc.odl.mit.edu/
STATIC_API_BASE_URL=https://ocwnext-rc.odl.mit.edu/
RESOURCE_BASE_URL=https://ocwnext-rc.odl.mit.edu
WWW_CONTENT_PATH=/home/gumaerc/Code/ocw-content-rc/ocw-www/
```
 - Make temporary edits to `course_carousel_card.html` and add extra test text after `{{ $courseTitle }}`, `{{ partial "course_cards_instructors.html" $courseInstructors }}` and `{{ partial "course_cards_topics.html" $courseTopics }}`, making sure you add enough text that it would cause the content to overflow the card
 - Run `npm run start:www` and visit http://localhost:3000
 - Make sure that none of the course cards overflow their bounds

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/12089658/173875489-69ba108b-538c-44a7-8f98-1b1605b886bd.png)
![image](https://user-images.githubusercontent.com/12089658/173875572-e330a326-89a9-4062-b78a-0ef4e71be5ef.png)
